File: shopapp-backend/shopapp/src/main/java/com/project/shopapp/dtos/CommentDTO.java
package com.project.shopapp.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.project.shopapp.models.Product;
import com.project.shopapp.models.User;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.stereotype.Component;

@Getter
@Setter
@Data
@Builder
@Component
@AllArgsConstructor
@NoArgsConstructor
public class CommentDTO {
    @JsonProperty("product_id")
    private Long productId;

    @JsonProperty("user_id")
    private Long userId;

    @JsonProperty("comment_content")
    private String commentContent;
    @JsonProperty("rating")
    private int rating;
    @JsonProperty("order_id")
    private Long orderId;
    @JsonProperty("is_active")
    private boolean active;
}

File: shopapp-angular-prefix/shopapp-angular/src/app/dtos/comment/comment.dto.ts
export class CommentDto{
  product_id: number;
  user_id: number;
  fullname: string;
  comment_content: string;
  rating: number;
  order_id: number;
  created_at: Date;

  constructor(data: any) {
    this.product_id = data.product_id;
    this.user_id = data.user_id;
    this.fullname = data.fullname;
    this.comment_content = data.comment_content;
    this.rating = data.rating;
    this.order_id = data.order_id;
    this.created_at = data.created_at;
  }
}
